make-rules/environment.mk

```diff
@@ -48,6 +52,63 @@ component-environment-prep::
 		{ echo "Adding required packages to build environment..."; \
 		  $(PFEXEC) /usr/bin/pkg install --accept -v $(REQUIRED_PACKAGES:%=/%) || [ $$? -eq 4 ] ; }
 
+ZONENAME_PREFIX = bz
+ZONENAME_ID = $(shell echo "$(WS_TOP)" | sha1sum | cut -c0-7)-$(COMPONENT_NAME)
+ZONENAME = $(ZONENAME_PREFIX)-$(ZONENAME_ID)
+
+component-zone-template:
+	$(call separator-line)
+	$(call separator-line,Create template zone)
+	USER_ID=$$(id -u) && \
+	$(PFEXEC) $(ZONE) --prefix $(ZONENAME_PREFIX) create-template -u $${USER} -i $${USER_ID}
+
+component-zone-build:
+	$(call separator-line)
+	$(call separator-line,Create $(ZONENAME))
+	$(PFEXEC) $(ZONE) --prefix $(ZONENAME_PREFIX) spawn-zone --id $(ZONENAME_ID)
+	$(call separator-line,Boot $(ZONENAME))
+	@while $$(true); do \
+		echo "Waiting for zone $(ZONENAME) to boot..."; \
+		$(PFEXEC) /usr/sbin/zlogin -l $${USER} $(ZONENAME) \
+				/bin/true >/dev/null 2>&1 && break; \
+		sleep 10; \
+	done
+	# FIXME:
+	# - remove once we figure out a better way how to enable zoneproxy-client inside the nlipkg brand
```

Comment:
Optionally we could teach sysding (or pkgcreatezone) to perform necessary actions. Perhaps, sysding is general enough. I don't insist on doing it now, just proposing.

Reply:
pkgcreatezone would be the place to do this, but I am not certain if it's a good idea to do this for all nlipkg zones or we shoudl introduce a new "nlipkg+zone-proxy" brand specific for just this use case.